ding/utils/log_writer_helper.py

```diff
@@ -0,0 +1,97 @@
+from tensorboardX import SummaryWriter
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from ding.framework import Task
+
+
+class DistributedWriter(SummaryWriter):
+    """
+    Overview:
+        A simple subclass of SummaryWriter that supports writing to one process in multi-process mode.
+        The best way is to use it in conjunction with the ``task`` to take advantage of the message \
+            and event components of the task (see ``writer.plugin``).
+    """
+
+    def __init__(self, *args, **kwargs):
+        self._default_writer_to_disk = kwargs.get("write_to_disk") if "write_to_disk" in kwargs else True
+        kwargs["write_to_disk"] = False
+        super().__init__(*args, **kwargs)
+        self._in_parallel = False
+        self._task = None
+        self._is_writer = False
+        self._lazy_initialized = False
+
+    def plugin(self, task: "Task", is_writer: False) -> "DistributedWriter":
+        """
+        Overview:
+            Plugin ``task``, so when using this writer in the task pipeline, it will automatically send requests\
+                to the main writer instead of writing it to the disk. So we can collect data from multiple processes\
+                and write them into one file.
+        Usage:
+            ``DistributedWriter().plugin(task, is_writer=("node.0" in task.labels))``
+        """
+        if task.router.is_active:
+            self._in_parallel = True
+            self._task = task
+            self._is_writer = is_writer
+            if is_writer:
+                self.initialize()
+            self._lazy_initialized = True
+            task.router.register_rpc("distributed_writer", self._on_distributed_writer)
+            task.once("exit", lambda: self.close())
+        return self
+
+    def _on_distributed_writer(self, fn_name: str, *args, **kwargs):
+        if self._is_writer:
+            getattr(self, fn_name)(*args, **kwargs)
+
+    def initialize(self):
+        self.close()
+        self._write_to_disk = self._default_writer_to_disk
+        self._get_file_writer()
+        self._lazy_initialized = True
+
+    def __del__(self):
```

Comment:
whether need to check some flag first to avoid calling close method repeatedly

Reply:
Double close check is done in the super method, so we don't need to check this